pkg/networkservice/mechanisms/vlan/client.go

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) 2021 Nordix Foundation.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package vlan
+
+import (
+	"context"
+	"io"
+	"time"
+
+	"git.fd.io/govpp.git/api"
+
+	interfaces "github.com/edwarnicke/govpp/binapi/interface"
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+	"github.com/networkservicemesh/api/pkg/api/networkservice/mechanisms/cls"
+	vlanmech "github.com/networkservicemesh/api/pkg/api/networkservice/mechanisms/vlan"
+	"github.com/networkservicemesh/api/pkg/api/networkservice/payload"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/chain"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
+	"github.com/networkservicemesh/sdk/pkg/tools/log"
+	"github.com/networkservicemesh/sdk/pkg/tools/postpone"
+
+	"github.com/networkservicemesh/sdk-vpp/pkg/networkservice/mechanisms/vlan/hwaddress"
+	"github.com/networkservicemesh/sdk-vpp/pkg/networkservice/mechanisms/vlan/l2vtr"
+	"github.com/networkservicemesh/sdk-vpp/pkg/networkservice/mechanisms/vlan/linkinit"
+	"github.com/networkservicemesh/sdk-vpp/pkg/tools/ifindex"
+)
+
+const (
+	serviceDomainLabel = "serviceDomain"
```

Comment:
Why are we doing this with a label rather than as part of the networkservice name?

Reply:
If the domain name is added to the service name the client must be configured to connect to this service by specifying the domain name also. The client configuration should not contain information about domain or vlan ID.

Reply:
@ljkiraly Network Service Mesh already has network service domain as part of the Network Service Name.

So for example if the client wants 'foo@radio.example.com' the annotation would be:

```
networkservicemesh.io: kernel://foo@radio.example.com
```

The user doesn't have to know anything about the vlan ID

Reply:
The NSE is running locally, and no other domain involved. So the service domain in my case is just a label. It is used to identify the physical device to use. What about renaming it to domain-label?